src/embed_tests/TestPythonEngineProperties.cs

```diff
@@ -183,15 +183,13 @@ public void SetPythonPath()
             PythonEngine.Initialize();
 
             const string moduleName = "pytest";
-            bool importShouldSucceed;
             try
             {
                 Py.Import(moduleName);
-                importShouldSucceed = true;
             }
-            catch
+            catch (PythonException ex)
             {
-                importShouldSucceed = false;
+                throw new Exception($"Py.Import(\"{moduleName}\") failed before setting PythonEngine.PythonPath", ex);
```

Comment:
You can't raise an exception here, because that would prevent `PythonEngine.Shutdown();` a few lines below from running, which brings it into inconsistent state.

Reply:
Technically, I am not raising a new exception, I am just appending to the message of an exception that was already raised.  There are many tests in that file that do not call Shutdown in case an exception is raised, e.g. GetPythonPathDefault, GetProgramNameDefault, GetPythonHomeDefault, SetPythonHomeTwice, SetProgramName.  Are you saying these are all wrong?

Reply:
I notice that a few tests in pyinitialize.cs also have this problem.  LoadDefaultArgs uses `using (new PythonEngine())` to handle Initialize/Shutdown.  Perhaps that is how all these other tests should be written?